daemon/api_debug_migrate.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var snapstateMigrateHome = snapstate.MigrateHome
+
+func migrateHome(st *state.State, snaps []string) Response {
+	if len(snaps) == 0 {
+		return BadRequest("no snaps were provided")
+	}
+
+	tss, err := snapstateMigrateHome(st, snaps)
+	if err != nil {
+		if terr, ok := err.(snap.NotInstalledError); ok {
+			return SnapNotFound(terr.Snap, err)
```

Comment:
Silly question - what happens here if multiple snaps are not found? Is err.Snap something like "snap1,snap2" then?

Reply:
No, it's just the first missing snap. We have the same issue in `snap refresh` and other operations on multiple snaps. This might be a nice first bug for new contributors since it's very simple but touches `snapstate`, `daemon` and `cmd/snap`.